keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1813 @@
+    
+      * Row 2 Annotation [b]: For container level resources, if a request is not
+        specified but a limit it set, the Pod Spec is modified to set the [request
+        equal to limit](https://github.com/kubernetes/kubernetes/blob/005f184ab631e52195ed6d129969ff3914d51c98/pkg/apis/core/v1/defaults.go#L167-L228) for each container. This modified Pod spec is then used by
+        the scheduler, and also for configuring cgroup settings.
+  
+  * **Rows 5 - 8**
+
+    ![](./pod-limit-set.png)
+    
+      * Row 5 Annotation [d]: When the pod-level request is not specified and
+        cannot be inferred from the container-level request because
+        container-level request is also unset, the pod-level request defaults to
+        the limit. 
+
+      * Row 7 Annotation [c]: When container-level limit is not set, the
+        pod-level limit is applied to each container's cgroup maximum value. This
+        is because container-level limit is implied in this case, and some
+        runtimes (like the Java runtime) rely on container-level cgroup maximum
+        values for fine tuning their components. This implied pod-level limit is
+        not set in Pod spec.  
+
+  * **Rows 9 - 12**
+
+    ![](./pod-request-set.png)
+
+      * When the pod-level limit is unset, the value will be derived from
+        container-level limits. If no pod-level limit is defined and not all the
+        containers within the pod have explicitly limits set, then the containers in the pod can
+        use full amount of resources available on the node, up to the node's
+        allocatable capacity.
+
+      * Row 10 Annotation [b]: For container level resources, if a request is not
+        specified but a limit it set, the Pod Spec is modified to set the [request
+        equal to limit](https://github.com/kubernetes/kubernetes/blob/005f184ab631e52195ed6d129969ff3914d51c98/pkg/apis/core/v1/defaults.go#L167-L228) for each container. This modified Pod spec is then used by
+        the scheduler, and also for configuring cgroup settings.  
+
+  * **Rows 13 - 16**
+
+    ![](./pod-request-limit-set.png)
+
+      * When both pod-level request and limit are explicitly specified,the scheduler and
+        cgroups will use those values directly.
+    
+      * Rows 13, 15 Annotation [c]: When container-level limit is not set, the
+        pod-level limit is applied to each container's cgroup maximum value. This
+        is because container-level limit is implied in this case, and some
+        runtimes (like the Java runtime) rely on container-level cgroup maximum
+        values for fine tuning their components.
+<br>
+
+##### Clarifying Tricky Cases and Design Decisions with Pod Level Resources
+
+1. [Invalid Pod Spec] Row 5 from the scenarios table: Pod Limits Defined; Pod
+   Requests & Container Specs Unspecified.
+   
+   ![](./example-1.png)
+  
+   * The Challenge of Missing Requests: In this case, Kubernetes cannot determine
```

Comment:
Given all we discussed today, would this not set PR from PL?

At L318 you said:

 "This means that at the pod level, if a resource request is not explicitly
   defined, Kubernetes attempts to derive it from requests specified
   at container-level within the pod. However, container-level requests are not
   set for at least 1 container, the pod-level request cannot be
   accurately derived. In this case,  the pod's resource request will default to
   the pod's limit, if pod limit is set."

Reply:
I think the pull request was updated after your comment or I was in middle of making that change. This is update now. Need to proof-read the entire KEP for missing details.